test/gtest/p4runtime.cpp

```diff
@@ -677,4 +678,54 @@ TEST_F(P4Runtime, StaticTableEntries) {
                 std::string("\x00\x00", 2), std::string("\x00\x04", 2));
 }
 
+TEST_F(P4Runtime, IsConstTable) {
+    auto test = createP4RuntimeTestCase(P4_SOURCE(P4Headers::V1MODEL, R"(
+        header Header { bit<8> hfA; }
+        struct Headers { Header h; }
+        struct Metadata { }
+
+        parser parse(packet_in p, out Headers h, inout Metadata m,
+                     inout standard_metadata_t sm) {
+            state start { transition accept; } }
+        control verifyChecksum(inout Headers h, inout Metadata m) { apply { } }
+        control egress(inout Headers h, inout Metadata m,
+                        inout standard_metadata_t sm) { apply { } }
+        control computeChecksum(inout Headers h, inout Metadata m) { apply { } }
+        control deparse(packet_out p, in Headers h) { apply { } }
+
+        control ingress(inout Headers h, inout Metadata m,
+                        inout standard_metadata_t sm) {
+            action a() { sm.egress_spec = 0; }
+            action a_with_control_params(bit<9> x) { sm.egress_spec = x; }
+
+            table t_const {
+                key = { h.h.hfA : exact; }
+                actions = { a; a_with_control_params; }
+                default_action = a;
+                const entries = {
+                    (0x01) : a_with_control_params(1);
+                }
+            }
+            table t_non_const {
+                key = { h.h.hfA : exact; }
+                actions = { a; a_with_control_params; }
+                default_action = a;
```

Comment:
The P4-16 spec says:  Future versions of P4 may add the ability to mix mutable and immutable entries in the same table, by declaring additional entries properties without the const keyword. Are you prepared for that case?

Reply:
That's a good point. We currently disallow that scenario - which is consistent with the language spec - and the "is_const_table" attribute in P4Info makes sense and is convenient, as it lets the application know whether are not it has write access to the table.

When - and if - we support mutable entries, this attribute will probably not make sense any more. However, I believe this will be a large change with other ramifications for us, so I believe it will probably translate to a new P4Runtime version in which we will deprecate that attribute (using the protobuf deprecation mechanism) and introduce a different one (different ones?) if needed.

Reply:
As long as there is something half-way plausible for a path to supporting that, _if_ it is ever needed, I think YAGNI applies here: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
I expect that there is a way which won't require deprecation. Const tables will always make sense. You will just have to add support for table initializers in some way: a table that starts with some entries populated by the P4 program - even though they can be modified later.